src/core/experimental/array.d

```diff
@@ -0,0 +1,1486 @@
+///
+module core.experimental.array;
+
+import core.experimental.refcount;
+
+import core.internal.traits : Unqual;
+
+/**
+Returns the size in bytes of the state that needs to be allocated to hold an
+object of type `T`. `stateSize!T` is zero for `struct`s that are not
+nested and have no nonstatic member variables.
```

Comment:
If that's your intention you need to special-case it below, since `T.sizeof` for a struct with no fields is always at least 1. Check it out:
https://run.dlang.io/is/USB7aW
```d
struct S {}
static assert(S.sizeof == 1);
void main()
{    
}
```

Reply:
Yes, this is currently not implemented. Thanks for reminding me, replacing this with a TODO.